### DISCUSSION TITLE
a view on how this could be

### DISCUSSION THREAD

**Discussion by ghost** [helloSystem/hello#295] [Ideas]
Good day, 

The following are some thougths on how I see this project, no specific order, and by no means in order of importance and might need some elaboration. Of course some of them might be already taken into account in the HIG and "wanted technologies", so take this just as a thinking aloud to talk about.

1-Icons in SVG all over. SVG icons scale without issues regardless the display resolution and user scaling preferences. SVG icons for apps, folders, devices... and also for toolbars.

2- Theming. Of course theming. Desktop look should be decided by the user, same as even we live in same floorplan flat, will have different colors on the walls and furnitures. Don't go that far as Plasma where everything is able to being colored in a different way, nor go as "far" as apple where only the highlight color can be set.
If there is going to be a dark theme, at least should be a "black" theme. Dark themes a la Apple or Adwaita are worst designs, ergonomically speaking, for eyes. Also allow to set a font for the system.

3- Straightforward multilanguage support. Adding an input method should be as easy as "InputMethod>Add new input method>Chinese (pinyin)" and that's it, can type chinese. And please, use standard ISO two characters country code for languages, and a third maybe, in the case that there is a variant like 'english international with alt gr dead keys', though last third char would need some debate on how to implement it.

4- Application's menu should indicate which application it belongs to. When multiple windows are open, there is no way to identify to which one the menu belongs to as there is no indication on to which one it belongs. That's why apple added the application name there. I personally don't like the global menu, though I can live with it if it gets improved. We could debate quite a long on the pros and cons of the global menu and small and big screens and windowed and full screen apps, though that is not the point here. Global menu is here to stay so IMHO it should be improved.

5- The fact that the user view of the system should be simple enough for the user not to feel scared nor overwhelmed does not mean that there should not be advanced features. Maybe could be interesting to add an "Advanced settings" for the power users hidden in the menu of the preferences, triggered when pressing a key and clicking on the menu? Needs some thought.

6- Sandboxed develop environment? I'm thinking about what Fedora has done with Silverblue/Kinoite. There is a toolbox (podman container iirc) where I can install all development stuff and the base system doesn't get polluted with development files. If I finish development, I remove the toolbox, and I'm back to a clean system without unneeded libraries, applications, modules, programs.... The system is always in a fresh state.

7- Standardize the preference applications' windows, using a "default" window for preferences. Therefore the user will know, just by looking at the window, that it's a preference and not other kind of software. Maybe could add an icon that indicates that the preference is set to defaults, and if the user changes it, the icon could change to indicate that it's a user set preference. (I'm working on a prototype to show this idea).

I will add more items here as they come to mind. I also have a view on how to organize the filesystem, but it doesn't make sense with the actual base system, so let's leave away for now.

Regards,
RR

**Comment by probonopd** [comment]
Hello @roired, thanks for your thoughtful comments.

As you have probably realized by now, helloSystem is not here to be yet another desktop environment or distribution that basically does what everyone does. Instead, we want to build a system that is simple yet powerful and welcoming to "mere mortals". This does require some opinionated decisions. To paraphrase Steve Jobs - you can please some people some of the time, but you cannot please all people all of the time.

> Icons in SVG

There are pros and cons to it. You are clearly aware of the pros, so let me give a con: Using the same vector image for all icon sizes does not always result in the best possible outcome. Especially for small icon sizes, one may want specially designed, simplified icons. The Mac uses icns, a format that can contain different icons for multiple resolutions within one single file. It is very elegant. Maybe we should make more use of it. Someone would need to put together icns icons for all applications in helloSystem. It's quite a bit of work, but a dedicated designer could certainly get this done. It would be no problem to support this format in Filer.

> Theming

helloSystem aspires to come with a distinctive user interface that should not be changed by random themes. Both Windows and the Mac did not have real theming for the first 2 or so decades of their existence (apart from changing some colors and backgrounds), and it resulted in a very uniform look and feel, where the screenshots in the manuals actually matched what you saw on screen. As soon as you allow theming, applications won't look as tested anymore, and the results will be unpredictable. For helloSystem we want to create a recognizable system that looks and works the same for everyone. For me the worst user interfaces are those where no 2 systems look and work the same, especially when I am the one who gets tech support calls over the phone. So while helloSystem doesn't restrict the user's power to use different themes, it's not something we currently encourage or expose in the GUI.

> Straightforward multilanguage support

This is a topic that I find really important, and most operating systems are _way_ too complicated in this area imho. I tend to really think in _simple_ terms. Like "English". If necessary, "English (US) and "English (UK)". I don't even have a desire to understand what "english international with alt gr dead keys" is (besides complicated techno jargon). Dead or alive, my keys need to "just work" (and work they do if they type what is printed on them).

Check out "Setting the keyboard language" in my [article](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-3-780f127f5794), and the videos therein. This is a topic that really can drive one nuts. I think it comes through in the videos, so entertaining to watch ;-)

When you run helloSystem on a Mac or on a system with a Raspberry Pi keyboard, the keyboard and system language will automatically be set correctly, with no manual user interaction at all. It fits my use case 100%. I have a German keyboard, so my system will be in German. Done. I realize that e.g., for Chinese the situation is not as straightforward but I lack the knowledge to create a similar "it just works" experience for e.g., Chinese. Maybe you can help [here](https://github.com/helloSystem/docs/issues/34)?

> And please, use standard ISO two characters country code for languages

Why not just use regular names? Codes are for computers. Humans tend to call languages "English", "Russian", "German",...

> The fact that the user view of the system should be simple enough for the user not to feel scared nor overwhelmed does not mean that there should not be advanced features

Agree.

This is why we currently have this - ugly and complicated as hell (I would argue _no one_ understands what it all means) but apparently it is useful for _some_:

![image](https://user-images.githubusercontent.com/2480569/146433260-a9747e4d-971c-4a11-a99e-917733946541.png)

I am not really at ease with this. This seems to be a dumping ground for Unix technical debt. Languages, keyboard layouts, countries, variants, all mixed together in a giant unexplained mess. The Mac apparently doesn't need such long lists of things no one understands. (Unfortunately I don't have a better solution at the moment, apart from disabling that menu again.)

> Sandboxed develop environment

For 0.7.0 we will not install the developer tools by default for the first time. They will be offered as a separate download. If you want to switch back and forth between a developer and a non-developer systems, you can use the __Boot Environments__ preferences application for this purpose. Just create a Boot Environment before you install the Developer Tools.

> Standardize the preference applications' windows, using a "default" window for preferences.

helloSystem is all about "software minimalism", we don't want to have more than what is needed:

![image](https://user-images.githubusercontent.com/2480569/146434453-a7be7774-51dd-4a93-a20a-30b37db6a0dc.png)

Overall, I think reading my series on #LinuxUsability ([part 1](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-1-5fa0fb369b42), [part 2](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-2-d34b86fd9b79) , [part 3](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-3-780f127f5794), [part 4](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-4-13be6486b358), [part 5](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-5-d810a0d2f368), [part 6](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-6-1c03de7c00a9)) will give a lot of relevant background regarding to the design intentions for helloSystem. 

Happy to discuss!
